hack/test_errata.py

```diff
@@ -0,0 +1,738 @@
+    def test_load_nonexisting_file(self):
+        """
+        Test loading a nonexisting file.
+        """
+        with tempfile.TemporaryDirectory() as tempdir:
+            cachepath = os.path.join(tempdir, "cache.json")
+            self.assertCountEqual(errata.load(cachepath), {})
+
+
+    def test_save_and_load_as_a_pair(self):
+        """
+        Test using errata.save and errata.load as a pair to confirm their functionality.
+        """
+        param_list = [
+            (),
+            ({"foo": "bar"}),
+            ({"value": "1234"}),
+            ({"company": "Red Hat"}),
+            ({"foo": "bar"}, {"value": "1234"}, {"errata": "1234"}),
+            ({"value": "1234"}, {"foo": "bar"}, {"errata": "1234"})
+        ]
+        for cache in param_list:
+            with self.subTest():
+                with tempfile.TemporaryDirectory() as tempdir:
+                    cachepath = os.path.join(tempdir, "cache.json")
+                    errata.save(cachepath, cache)
+                    self.assertCountEqual(errata.load(cachepath), cache)
+
+
+class PollTest(unittest.TestCase):
+    def setUp(self):
+        self.raw_messages_containing_msg_expected_to_be_polled = [
+            {
+                "additional_unnecessary_info": "shouldn't be processed",
+                "msg": {
+                    "errata_id": 11,
+                    "product": "RHOSE",
+                    "to": "SHIPPED_LIVE",
+                }
+            },
+            {
+                "additional_unnecessary_info": "shouldn't be processed",
+                "msg": {
+                    "errata_id": 12,
+                    "product": "RHOSE",
+                    "to": "SHIPPED_LIVE",
+                }
+            }
+        ]
+        self.raw_messages_containing_msg_expected_not_to_be_polled = [
+            {
+                "additional_unnecessary_info": "shouldn't be processed",
+                "msg": {
+                    "errata_id": 21,
+                    "product": "RHOSE",
+                    "to": "QE",
+                }
+            },
+            {
+                "additional_unnecessary_info": "shouldn't be processed",
+                "msg": {
+                    "errata_id": 22,
+                    "product": "RHEL",
+                    "to": "SHIPPED_LIVE",
+                }
+            },
+            {
+                "additional_unnecessary_info": "shouldn't be processed",
+                "msg": {
+                    "errata_id": 23,
+                    "product": "RHEL",
+                    "to": "QE",
+                }
+            },
+            {
+                "additional_unnecessary_info": "shouldn't be processed",
+                "msg": {
+                    "errata_id": 24,
+                    "product": "SHIPPED_LIVE",
+                    "to": "RHOSE",
+                }
+            }
+        ]
+        self.multiple_raw_messages =                                    \
```

Comment:
A bit unreadable, maybe
```suggestion
        self.raw_messages = [
            (True, {
                    "additional_unnecessary_info": "shouldn't be processed",
                    "msg": {
                        "errata_id": 11,
                        "product": "RHOSE",
                        "to": "SHIPPED_LIVE",
                    }}),
            (True, {
                    "additional_unnecessary_info": "shouldn't be processed",
                    "msg": {
                        "errata_id": 12,
                        "product": "RHOSE",
                        "to": "SHIPPED_LIVE",
                    }}),
            (False, {
                    "additional_unnecessary_info": "shouldn't be processed",
                    "msg": {
                        "errata_id": 24,
                        "product": "SHIPPED_LIVE",
                        "to": "RHOSE",
                    }})
        ]
        self.valid_messages = [x[1] for x in self.raw_messages if x[0]]
```
would be a bit better?

Reply:
I must have overlooked the backslash when I was double-checking the code, thank you.